### ISSUE TITLE
`frontend:dev:watcher` should be executed after `frontend:install`.

### ISSUE THREAD

**Issue by misitebao** [wailsapp/wails#1182]
**Description**
After creating the project, run `wails dev` for the first time. When `frontend:install` has not been executed, `frontend:dev:watcher` will also be executed, and the system prompts lack of dependencies.

**To Reproduce**
Steps to reproduce the behaviour:
1. Generate `wails cli` with `master`  branch
2. Run ` wails init -n wails-vue -t https://github.com/misitebao/wails-template-vue`
3. Add `"frontend:dev:watcher": "npm run build:watch -w js",` to `wails.json` file
4. Run `wails dev`

**Expected behaviour**
Both synchronous and asynchronous scripts should be executed after the `frontend:install` command. Otherwise there will be missing dependencies.

**Screenshots**
![image](https://user-images.githubusercontent.com/28185258/155425320-f68bf2a3-8d1c-4131-806a-1d42403c49c2.png)

**System Details**
```shell
$ wails doctor
Wails CLI v2.0.0-beta.32

Scanning system - Please wait (this may take a long time)...Done.

System
------
OS:             Windows 10 Pro
Version:        2009 (Build: 19044)
ID:             21H2
Go Version:     go1.17.6
Platform:       windows
Architecture:   amd64

Dependency      Package Name    Status          Version
----------      ------------    ------          -------
WebView2        N/A             Installed       98.0.1108.56
npm             N/A             Installed       8.5.1
*upx            N/A             Available

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!
Optional package(s) installation details:
  - upx : Available at https://upx.github.io/

If Wails is useful to you or your company, please consider sponsoring the project:
https://github.com/sponsors/leaanthony

```

**Additional context**
None

**Comment by misitebao** [comment]
@stffabi 
The previous test used an existing project, and the dependencies have been installed, so this problem was not found. I just created a new project and found this problem during the test.

**Comment by stffabi** [comment]
I would be fine calling an install before starting the frontend watcher. To be honest I have no expertise with these integrations, as we currently use another approach for integrating the frontend into wails. If @leaanthony is fine calling an install there I could provide a PR for this.

**Comment by leaanthony** [comment]
I just feel that if you're spinning out a watcher, then that should probably handle everything involved in watching. Is that how they work? I intend to move the svelte template to vite so I'll get a bit more insight into this. Thanks @stffabi for offering to do the install. I really want to understand the use cases here for external watcher and proxying to Dev servers like vite. Everyone wants this!

**Comment by stffabi** [comment]
Yeah, absolutely. Unfortunately I really have no expertise and I don't have any usecase for this. So I'm not really a help...

**Comment by stffabi** [comment]
> I just feel that if you're spinning out a watcher, then that should probably handle everything involved in watching. Is that how they work?

I had the same feeling.

**Comment by leaanthony** [comment]
Amazing! Thanks @misitebao

**Comment by misitebao** [comment]
@stffabi @leaanthony 
This issue may need to be discussed, because `frontend:dev:watcher` and `wails dev` are asynchronous, but both need to execute the `fontend:install` command, the former needs to be executed after the `fontend:install` command is completed. To enable `watcher`. Not to mention the latter, front-end resource builds must execute `frontend:install` before proceeding to subsequent steps. We may need to make some adjustments to the timing of `frontend:install` to make it a prerequisite for `frontend:dev:watcher` and `wails dev`.

**Comment by leaanthony** [comment]
Fixed by #1216
